### ISSUE TITLE
[AUSF] Request for Private-Key configuration details.

### ISSUE THREAD

**Issue by NeelimaKalpana** [magma/magma#10400]
This is w.r.t SUPI registration with Profile-A and Profile-B (TC1x and TC1y) testcases in TVM.
There we need to have public-key and private-key for profile-A and Profile-B .But I can see only Public-key parameter.
Please help me to find out the Private-Key configuration in TeraVM .

Screen Shot for Public-Key.

This is w.r.t SUPI registration with Profile-A and Profile-B (TC1x and TC1y) testcases in TVM.
There we need to have public-key and private-key for profile-A and Profile-B .But I can see only Public-key parameter.
Please help me to find out the Private-Key configuration in TeraVM .

Screen Shot for Public-Key.
![image.png](https://images.zenhubusercontent.com/6139f1931bcb9da2af1bf651/28475632-ab1a-468e-8a0f-2267e3e35596)

Kindly please let me know where I can find private-Key parameters, if that is not existing kindly add the same in the TVM configurations.

**Comment by arun-magma** [comment]
will check this

**Comment by NeelimaKalpana** [comment]
Okay Arun , please check and let me know.

**Comment by NeelimaKalpana** [comment]
Hi Arun,

please let me know the status of adding private-key parameter in TVM.

**Comment by arun-magma** [comment]
Private Key is not needed for SUCI encryption both at SIM/UE side, we can use public key at UE/SIM  side to conceal the SUCI & go with the Reg process in AGW private key can be used for deconceal it, Kindly let us know the significance if any if magma differs with this logic.

![Screen Shot 2021-11-22 at 11.55.30 PM.png](https://images.zenhubusercontent.com/5f934b19f9d04269a3f065cd/2019c890-01fb-4eea-83a4-92a85ee89c69)

**Comment by NeelimaKalpana** [comment]
Hi Arun,

I will discuss with developers and give and update on significance of Private-Key.

**Comment by Kaleem-Wavelabs** [comment]
@arun-magma @NeelimaKalpana 
Would like to know how UE specific public key is generated from test scripts.
As we know, home key identifier which represents unique public/private key is used to generate UE specific public key and encrypt IMSI

At AGW side, with UE provided public key and home key identifier, it will deconceal IMSI

This is the sample output for encryption and decryption logic:

 ==== KEY GENERATION in AGW ====
ProfileA
Public key: b'\xd5>\x91\xb5"kl\xb3ad\x05j%N\x89\xb2\x9dx\xec\xc962n6\x05 `\x9f\x84\x1fT"' 32
Private key: b'\xc0t\x17\x047\xc3\xeb\xa5\x02a\x12QO\x11\xb8\xe6$~\xf4\xef\x87|\xc0\xff\xe0\xecR\xf9\xb7+\x87e' 32

 ==== ENCRYPTION in UE ====
U̲E̲ ̲:̲ ̲U̲s̲i̲n̲g̲ ̲h̲o̲m̲e̲ ̲n̲e̲t̲w̲o̲r̲k̲ ̲p̲u̲b̲l̲i̲c̲ ̲k̲e̲y̲ ̲g̲e̲n̲e̲r̲a̲t̲e
ue pub_key: b'<\x15\xe3J\xe9W\xac\xceP\xcd\xae\x8fn&g\x87\xa6\x01:\xbf\x92)\xebS\x07\xf5\xfdo3I\xf1\x03',
cipher_tex: 849df42060,
mac: f6d157e22999d685

 ==== DECRYPTION in AGW ====

O̲U̲T̲P̲U̲T̲ ̲O̲F̲ ̲M̲S̲I̲N̲ ̲C̲O̲M̲P̲A̲R̲E
UE MSIN        =  1032547698

**Comment by NeelimaKalpana** [comment]
Hi Arun,

**Comment by NeelimaKalpana** [comment]
Hi Arun,

This is not closed. By mistake I clicked on close.
Please give clarity on public-key generation .

**Comment by arun-magma** [comment]
@Kaleem-Wavelabs @NeelimaKalpana : As discussed options are available in UE profile config to add public key id, Public key and Private key (optional/trivial) for Profile-A and Profile-B, this configs would reflect only in TC1x and TC1y related scripts.

![Screen Shot 2021-11-23 at 8.55.13 AM.png](https://images.zenhubusercontent.com/5f934b19f9d04269a3f065cd/448dcbaf-f9fa-4391-a7be-1b92f2c37282)

**Comment by NeelimaKalpana** [comment]
Hi Arun,

Thank you for doing needful.
Once code is ready I will Test it and give an update on the TC1x and TC1y functionality.

**Comment by arun-magma** [comment]
I'm closing this task, if any more info needed please open a new ticket. Thanks!
